### PR TITLE
Add override_gem option to Gemfile

### PR DESCRIPTION
**Pull request by agrare**
Add the ability for developers to override gems in the Gemfile with
local repositories to aid in testing changes out that span multiple
repositories.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -40,3 +40,30 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
...
 gem 'rest-client'
 gem 'rspec-rails'
+
+#
+# Custom Gemfile modifications
+#
+# To develop a gem locally and override its source to a checked out repo
+#   you can use this helper method in Gemfile.dev.rb e.g.
+#
+# override_gem 'manageiq-ui-classic', :path => "../manageiq-ui-classic"
```
Comment (bdunne) [inline Gemfile]: The comment should probably be updated to use `topological_inventory-core`.

File: Gemfile [hunk 2]
```diff
@@ -40,3 +40,30 @@ gem 'tzinfo-data', platforms: [:mingw, :mswin, :x64_mingw, :jruby]
 
 gem 'rest-client'
 gem 'rspec-rails'
+
+#
+# Custom Gemfile modifications
+#
+# To develop a gem locally and override its source to a checked out repo
+#   you can use this helper method in Gemfile.dev.rb e.g.
```
Comment (bdunne) [inline Gemfile]: Gemfile.dev.rb should probably be bundler.d/*.rb

File: Gemfile (hunk 1 above)
Comment (agrare) [reply Gemfile]: :+1: done

File: Gemfile (hunk 2 above)
Comment (agrare) [reply Gemfile]: yeah, done

**Pull request merged**